src/test/java/com/uid2/shared/store/reader/RotatingKeysetProviderTest.java

```diff
@@ -0,0 +1,140 @@
+
+        content.add(entry);
+    }
+
+    private JsonObject makeMetadata(String location) {
+        JsonObject metadata = new JsonObject();
+        JsonObject keysets = new JsonObject();
+        keysets.put("location", location);
+        metadata.put("keysets", keysets);
+        return metadata;
+    }
+
+    @Test
+    public void loadsContentToSiteScope() throws Exception{
+        RotatingKeysetProvider provider = new RotatingKeysetProvider(cloudStorage, new SiteScope(new CloudPath("metadata"), 5));
+        JsonArray content = new JsonArray();
+        addKeyset(content, 5, 1, 1, 2, 3);
+        addKeyset(content, 1, 2, 1,2,5);
+        when(cloudStorage.download("locationPath")).thenReturn(makeInputStream(content));
+        final long count = provider.loadContent(makeMetadata("locationPath"));
+        Assert.assertEquals(2, count);
+    }
+
+    @Test
+    public void loadContentEmptyArray() throws Exception {
+        JsonArray content = new JsonArray();
+        when(cloudStorage.download("locationPath")).thenReturn(makeInputStream(content));
+        final long count = keysetProvider.loadContent(makeMetadata("locationPath"));
+        Assert.assertEquals(0, count);
+    }
+
+    @Test
+    public void loadsContent() throws Exception {
+        InMemoryStorageMock cloudStorage = new InMemoryStorageMock();
+        String contentPath = "file.json";
+        String metadataPath = "metadata.json";
+
+        cloudStorage.upload(toInputStream(makeMetadata(contentPath).encodePrettily()), metadataPath);
+
+        JsonArray content = new JsonArray();
+        addKeyset(content, 1, 1, 1, 2,3);
+        cloudStorage.upload(toInputStream(content.encodePrettily()), contentPath);
+
+        RotatingKeysetProvider provider = new RotatingKeysetProvider(cloudStorage, new GlobalScope(new CloudPath(metadataPath)));
+        provider.loadContent();
+
+        KeysetSnapshot snapshot = provider.getSnapshot();
+        Map<Integer, Keyset> actual = snapshot.getAllKeysets();
+        Set<Integer> expectedAllowList = new HashSet<>();
+        expectedAllowList.add(1);
+        expectedAllowList.add(2);
+        expectedAllowList.add(3);
+        assertThat(actual).hasSize(1);
+        assertThat(actual.get(1).getAllowedSites()).isEqualTo(expectedAllowList);
+
+    }
+    private ClientKey makeClientKey(int siteId) {
+        return new ClientKey("test-client-key", Utils.toBase64String("test-client-secret".getBytes(StandardCharsets.UTF_8))).withSiteId(siteId);
+    }
+    private KeysetKey makeKey(int keysetId) {
+        return new KeysetKey(0, null, null, null, null, keysetId);
+    }
+    private boolean canAccessKey(int clientSiteId, int keySiteId) {
+        return keysetProvider.getSnapshot().canClientAccessKey(makeClientKey(clientSiteId), makeKey(keySiteId));
+    }
+
+    @Test
+    public void loadContentMultipleEntries() throws Exception {
+        JsonArray content = new JsonArray();
+        addKeyset(content, 1, 1, 2,3);
```

Comment:
- should we add tests where allowed sites is empty?
- a test where an allowed sites list has duplicate site ids?